cerberus-dashboard/src/actions/authenticationActions.js

```diff
@@ -245,7 +245,44 @@ export function triggerCodeChallenge(mfaDeviceId, stateToken) {
                         </div>
                     </div>
                 ));
+                dispatch(resetAuthState());
+            });
+    };
+}
+
+/**
+ * This action triggers a push notification challenge for a user with MFA, if MFA is required
+ * @param mfaDeviceId ID of the MFA security device
+ * @param stateToken Identifying token for the authentication request
+ */
+export function triggerPushChallenge(mfaDeviceId, stateToken) {
+    return function (dispatch) {
+        dispatch(loginMfaChallenge());
+        return axios({
+            method: 'post',
+            url: environmentService.getDomain() + cms.USER_AUTH_MFA_PATH,
+            data: {
+                state_token: stateToken,
+                device_id: mfaDeviceId,
+                is_push: true
+            },
+            timeout: AUTH_ACTION_TIMEOUT
+        })
+            .then(function (response) {
+                handleUserLogin(response, dispatch);
+            })
+            .catch(function ({ response }) {
+                log.error('Failed to finalize MFA login', response);
```

Comment:
I tested this, and response is undefined when I let the push notification time out, not sure if that the expected behavior or not.

A better error message might be helpful here, maybe saying something about how it could be due to timeout. It threw a 504, which implies server error, but it was because I didn't click on the push notification which could be seen as a client error.

Reply:
good catch, I'll see about adding some better error handling here.

Reply:
Did a lot of updates to errors. should now give an unauthorized error code with a timeout and inform the user of why it happened.